File: GameHive/Constants/GameModeEnum.cs
/*************************************************************************************
 * 文 件 名:   GameModeEnum.cs
 * 描    述: 
 * 版    本：  V3.0 引入DRL、MinMax-MCTS混合算法
 * 创 建 者：  Cassifa
 * 创建时间：  2025/5/25 22:14
*************************************************************************************/
namespace GameHive.Constants.GameModeEnum {
    public enum GameMode {
        LocalGame,           // 本地对战
        LMMGame,             // 与大模型对战
        OnlineGame,         // 联机对战
    }

    public static class GameModeExtensions {
        public static string GetChineseName(this GameMode mode) => mode switch {
            GameMode.LocalGame => "本地对战",
            GameMode.LMMGame => "与大模型对战",
            GameMode.OnlineGame => "联机对战",
            _ => "未知模式"
        };

        public static string GetEnglishName(this GameMode mode) => mode switch {
            GameMode.LocalGame => "Local Game",
            GameMode.LMMGame => "AI Game",
            GameMode.OnlineGame => "Online Game",
            _ => "Unknown Mode"
        };
    }
}

File: Backend/gamehive-admin/src/main/java/com/gamehive/constants/GameModeEnum.java
package com.gamehive.constants;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 游戏模式枚举
 *
 * @author Cassifa
 */
@Getter
@AllArgsConstructor
public enum GameModeEnum {
    LOCAL_GAME(0, "本地对战", "Local Game"),
    LMM_GAME(1, "与大模型对战", "AI Game"),
    ONLINE_GAME(2, "联机对战", "Online Game");

    private final int code;
    private final String chineseName;
    private final String englishName;

    /**
     * 根据代码获取枚举
     *
     * @param code 代码
     * @return 对应的枚举值
     * @throws IllegalArgumentException 如果找不到匹配的枚举值
     */
    public static GameModeEnum fromCode(int code) {
        for (GameModeEnum value : values()) {
            if (value.getCode() == code) {
                return value;
            }
        }
        throw new IllegalArgumentException("未知游戏模式: " + code);
    }

    /**
     * 根据中文名称获取枚举
     *
     * @param chineseName 中文名称
     * @return 对应的枚举值
     * @throws IllegalArgumentException 如果找不到匹配的枚举值
     */
    public static GameModeEnum fromChineseName(String chineseName) {
        for (GameModeEnum value : values()) {
            if (value.getChineseName().equals(chineseName)) {
                return value;
            }
        }
        throw new IllegalArgumentException("未知游戏模式: " + chineseName);
    }

    /**
     * 根据英文名称获取枚举
     *
     * @param englishName 英文名称
     * @return 对应的枚举值
     * @throws IllegalArgumentException 如果找不到匹配的枚举值
     */
    public static GameModeEnum fromEnglishName(String englishName) {
        for (GameModeEnum value : values()) {
            if (value.getEnglishName().equals(englishName)) {
                return value;
            }
        }
        throw new IllegalArgumentException("未知游戏模式: " + englishName);
    }
} 